src/mrinufft/operators/interfaces/gpunufft.py

```diff
@@ -511,12 +511,14 @@ def samples(self, samples):
         samples: np.ndarray
             The samples for the Fourier Operator.
         """
+        self._samples = samples
```

Comment:
This was a bug we missed earlier. We used to compute density without updating samples. Fixed this now. However, I added a TODO that we still have performance issue, this needs gpuNUFFT to sort the samples twice, once for pipe and once for the operator. While we do have `pipe` method. It currently is messy to just call it directly (as we have `osf` and `upsampfac` as 2 different parameters. Also, we still create an operator to ensure we can call pipe without user needing to create a new fourier_op. @paquiteau whats the best way forward here? Lets discuss this sometime as we have all the elements to get rid of perf issues but we need to agree on standard to follow. This can greatly benefit in learning trajectory

https://github.com/mind-inria/mri-nufft/blob/20a77aaeee0a2ac71002c92d99d59a8650ba474d/src/mrinufft/operators/interfaces/gpunufft.py#L551-L557